pkg/knowledge_base2/reader/properties.go

```diff
@@ -0,0 +1,267 @@
+package reader
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/google/uuid"
+	construct "github.com/klothoplatform/klotho/pkg/construct2"
+	knowledgebase "github.com/klothoplatform/klotho/pkg/knowledge_base2"
+	"github.com/klothoplatform/klotho/pkg/knowledge_base2/properties"
+	"gopkg.in/yaml.v3"
+)
+
+type (
+	Properties map[string]*Property
+
+	Property struct {
+		Name string `json:"name" yaml:"name"`
+		// Type defines the type of the property
+		Type string `json:"type" yaml:"type"`
+
+		Namespace bool `json:"namespace" yaml:"namespace"`
+
+		DefaultValue any `json:"default_value" yaml:"default_value"`
+
+		Required bool `json:"required" yaml:"required"`
+
+		ConfigurationDisabled bool `json:"configuration_disabled" yaml:"configuration_disabled"`
+
+		DeployTime bool `json:"deploy_time" yaml:"deploy_time"`
+
+		OperationalRule *knowledgebase.PropertyRule `json:"operational_rule" yaml:"operational_rule"`
+
+		Properties Properties `json:"properties" yaml:"properties"`
+
+		MinLength *int `yaml:"min_length"`
+		MaxLength *int `yaml:"max_length"`
+
+		LowerBound *float64 `yaml:"lower_bound"`
+		UpperBound *float64 `yaml:"upper_bound"`
+
+		AllowedTypes construct.ResourceList `yaml:"allowed_types"`
+
+		SanitizeTmpl  string   `yaml:"sanitize"`
+		AllowedValues []string `yaml:"allowed_values"`
+
+		KeyProperty   knowledgebase.Property `yaml:"key_property"`
+		ValueProperty knowledgebase.Property `yaml:"value_property"`
+
+		ItemProperty knowledgebase.Property `yaml:"item_property"`
```

Comment:
What are these for? They seem like they're computed but the yaml tag suggests otherwise. (check other fields too)

Reply:
if you want your item/key/value to be able to have validation along with it (the strings in my list need to match a regex) these would be where you would define that

Reply:
Doesn't seem like these should be full `knowledgebase.Property` then, since it's just validation and none of the other fields should be specified (ie, can a `key_property` have sub properties?).